app/src/main/java/io/appium/settings/location/LocationBuilder.java

```diff
@@ -55,8 +56,16 @@ public static Location buildFromIntent(Intent intent, String providerName) {
         Double altitude = extractParam(intent, ALTITUDE_PARAMETER_KEY);
         Double speed = extractParam(intent, SPEED_PARAMETER_KEY);
         Double bearing = extractParam(intent, BEARING_PARAMETER_KEY);
+        Double accuracy = extractParam(intent, ACCURACY_PARAMETER_KEY);
         Location location = new Location(providerName);
-        location.setAccuracy(Criteria.ACCURACY_FINE);
+
+        if (accuracy != null) {
```

Comment:
`location.setAccuracy(accuracy == null ? accuracy.floatValue() : Criteria.ACCURACY_FINE)` would be shorter